From e0bfc286af46e517ea9d715a3066778427a882ef Mon Sep 17 00:00:00 2001
From: Prarit Bhargava <prarit@redhat.com>
Date: Tue, 26 Aug 2014 12:49:19 -0400
Subject: [tools] turbostat: Check return value of fscanf

Message-id: <1409057382-26432-9-git-send-email-prarit@redhat.com>
Patchwork-id: 89090
O-Subject: [RHEL7.1 PATCH BZ 1133893 08/31] turbostat: Check return value of fscanf
Bugzilla: 1133893
RH-Acked-by: Dean Nelson <dnelson@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1133893

commit 74823419761c11830ea1819365f82cf3d48795cb
Author: Josh Triplett <josh@joshtriplett.org>
Date:   Tue Aug 20 17:20:15 2013 -0700

    turbostat: Check return value of fscanf

    Some systems declare fscanf with the warn_unused_result attribute.  On
    such systems, turbostat generates the following warnings:

    turbostat.c: In function 'get_core_id':
    turbostat.c:1203:8: warning: ignoring return value of 'fscanf', declared with attribute warn_unused_result [-Wunused-result]
    turbostat.c: In function 'get_physical_package_id':
    turbostat.c:1186:8: warning: ignoring return value of 'fscanf', declared with attribute warn_unused_result [-Wunused-result]
    turbostat.c: In function 'cpu_is_first_core_in_package':
    turbostat.c:1169:8: warning: ignoring return value of 'fscanf', declared with attribute warn_unused_result [-Wunused-result]
    turbostat.c: In function 'cpu_is_first_sibling_in_core':
    turbostat.c:1148:8: warning: ignoring return value of 'fscanf', declared with attribute warn_unused_result [-Wunused-result]

    Fix these by checking the return value of those four calls to fscanf.

    Signed-off-by: Josh Triplett <josh@joshtriplett.org>
    Signed-off-by: Len Brown <len.brown@intel.com>

Cc: Tony Camuso <tcamuso@redhat.com>
Cc: Dean Nelson <dnelson@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/power/x86/turbostat/turbostat.c b/tools/power/x86/turbostat/turbostat.c
index 14de623..4e77f2c 100644
--- a/tools/power/x86/turbostat/turbostat.c
+++ b/tools/power/x86/turbostat/turbostat.c
@@ -1207,7 +1207,10 @@ int cpu_is_first_sibling_in_core(int cpu)
   perror(path);
   exit(1);
  }
- fscanf(filep, "%d", &first_cpu);
+ if (fscanf(filep, "%d", &first_cpu) != 1) {
+  perror(path);
+  exit(1);
+ }
  fclose(filep);
  return (cpu == first_cpu);
 }
@@ -1228,7 +1231,10 @@ int cpu_is_first_core_in_package(int cpu)
   perror(path);
   exit(1);
  }
- fscanf(filep, "%d", &first_cpu);
+ if (fscanf(filep, "%d", &first_cpu) != 1) {
+  perror(path);
+  exit(1);
+ }
  fclose(filep);
  return (cpu == first_cpu);
 }
@@ -1245,7 +1251,10 @@ int get_physical_package_id(int cpu)
   perror(path);
   exit(1);
  }
- fscanf(filep, "%d", &pkg);
+ if (fscanf(filep, "%d", &pkg) != 1) {
+  perror(path);
+  exit(1);
+ }
  fclose(filep);
  return pkg;
 }
@@ -1262,7 +1271,10 @@ int get_core_id(int cpu)
   perror(path);
   exit(1);
  }
- fscanf(filep, "%d", &core);
+ if (fscanf(filep, "%d", &core) != 1) {
+  perror(path);
+  exit(1);
+ }
  fclose(filep);
  return core;
 }
-- 
1.7.1